mars/tensor/reshape/reshape.py

```diff
@@ -457,7 +458,17 @@ def _execute_map(cls, ctx, op):
                 t, target_chunk_idx[idx] = divmod(t, dim_chunk_count)
             target_chunk_idx.reverse()
 
-            ctx[chunk.key, tuple(target_chunk_idx)] = group_indices + (group_data,)
+            mapper_outputs[chunk.key, tuple(target_chunk_idx)] = group_indices + (
+                group_data,
+            )
+
+        # ensure all mapper data are inserted context in order and fill missing partition with None
```

Comment:
This adds more costs to the Mars backend, can we move this to Ray executor?

Reply:
Actually the cost is minimal. Mars shuffle fetch will filter out all `None` values.

Reply:
Yes, `None` values will be put into storage in Mars backend. But shuffle mapper meta are not recorded in mars, so the cost will be negligible？